packages/hooks/src/server/site-router.ts

```diff
@@ -0,0 +1,116 @@
+import type { Handle, RequestEvent } from '@sveltejs/kit';
+
+/**
+ * The uri object with site and entry properties.
+ */
+export type SiteRouter = {
+	default?: true;
+	site: {
+		uri: string;
+		handle: string;
+	};
+	entry: {
+		uri: string;
+	};
+};
+
+/**
+ * Options for the siteRouter handle.
+ */
+export type SiteRouterHandleOptions = {
+	defaultSiteUri: string;
+	defaultEntryUri?: string;
+	siteHandle?: (event: RequestEvent) => string;
+	pathnameSplitter?: (pathname: string) => string[];
+	partsToSiteRouterObject?: (parts: string[]) => SiteRouter;
+};
+
+/**
+ * Let's make sure the locals object has the uri object with site and entry properties.
+ */
+type SiteRouterLocals = App.Locals & {
+	siteRouter: SiteRouter;
+};
+
+/**
+ * Split the pathname into site and entry parts.
+ * This default implementation splits the pathname by the '/' character and filters out empty strings.
+ * @param pathname The pathname to split.
+ * @returns The site and entry parts.
+ */
+export const defaultPathnameSplitter = (pathname: string) => pathname.split('/').filter((x) => !!x);
+
+/**
+ * Convert the parts into a uri object.
+ * This default implementation takes the first part as the site and the rest as the entry.
+ * @param parts The parts to convert.
+ * @returns The uri object.
+ */
+export const defaultPartsToSiteRouterObject = (parts: string[]) =>
+	({
+		site: {
+			uri: parts[0],
+			handle: ''
+		},
+		entry: {
+			uri: parts.slice(1).join('/')
+		}
+	}) satisfies SiteRouter;
+
+/**
+ * Default site handle formatter.
+ * This default implementation replaces all '-' characters with '_' characters.
+ * @param siteRouter The site router object.
+ * @returns The formatted site handle.
+ */
+export const defaultSiteHandle = (siteRouter: SiteRouter) => {
+	return siteRouter.site.uri.replaceAll('-', '_');
+};
+
+/**
+ * This handle is responsible for setting the siteRouter object in the locals object.
+ * Its responsibility is to split the pathname into site and entry parts and convert
+ * them into a siteRouter object. It also formats the site handle.
+ *
+ * Many parts of the implementation are customizable through the options.
+ *
+ * Those values will be used by other hooks and should also be used when requesting
+ * an entry from the CMS.
+ *
+ * @param options The options for the siteRouter handle.
+ * @returns The siteRouter handle.
+ */
+export const createSiteRouter: (options: SiteRouterHandleOptions) => Handle = ({
+	defaultSiteUri,
+	defaultEntryUri = '',
+	siteHandle = defaultSiteHandle,
+	pathnameSplitter = defaultPathnameSplitter,
+	partsToSiteRouterObject = defaultPartsToSiteRouterObject
+}) => {
+	return <L extends SiteRouterLocals>({ event, resolve }) => {
+		const url = new URL(event.url);
```

Comment:
`event.url` is already a `URL` object. Is there a reason to parse it again?

Reply:
@f-elix Good catch. See 0a690c522b